src/ray/protobuf/gcs_service_autoscaler_dependencies.proto

```diff
@@ -0,0 +1,92 @@
+/* Please do not modify/remove/change the messages in this file to maintain
```

Comment:
can we just name this file `autoscaler.proto`? Also can we move the other dependencies from `common.proto` and `gcs.proto` here?

Reply:
I'm going to revert the proto file structure to what it was before, since changing this changes the structure of the generated files and this requires more code changes.

Reply:
Why not create `autoscaler.proto`? Other than actually testing the rpcs themselves (which i agree is hard), separating the code that needs to be compatible is probably the most important thing. otherwise we're just going to have to add more people to the codeowner and create more work for unrelated prs (which makes it more likely for people to miss issues).

Reply:
Making an `autoscaler.proto` sounds reasonable -- but some questions:
(1) Would the InternalKV interface also go in `autoscaler.proto`?
(2) Where does GcsStatus message go (it's used in services both in the autoscaler and non-autoscaler parts)
(3) Would this cause version incompatibilities before and after the file split?
...(4) Would it have made more sense in the first place to do one service per proto file?

I think the main thing my questions point to is that someone else (e.g. you) should do this reorganization :p

Reply:
> (1) Would the InternalKV interface also go in autoscaler.proto?
Yeah
> (2) Where does the GcsStatus message go (it's used in services both in the autoscaler and non-autoscaler parts)
In the autoscaler part. I think nearly all of the autoscaler interface is actually used for other internal communication. 

> (3) Would this cause version incompatibilities before and after the file split?

I don't think that information is encoded in the protobuf. 

> (4) Would it have made more sense in the first place to do one service per proto file?

Maybe, but you've gotta remember that the autoscaler wasn't originally built for k8s and compatibility. 

I don't have the bandwidth to work on this right now. @ericl do you want to chime in on if this should go on the core backlog.

Reply:
Okidoke, I'll do a quick check with a test python rpc client/server setup to see if a name discrepancy in proto files between client and server messes any thing up -- if that works out ok, will make an attempt at reorganizing the protos and find-and-replacing all the references to generated code on client and server side.

Reply:
Ok, yes indeed `helloworld.proto` on client side and `helloworld2.proto` on server side work fine together.